sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -169,7 +171,16 @@ yield visitBinaryArithmetic(
   }
 
   protected String visitLiteral(Literal<?> literal) {
-    return literal.toString();
+    String litString = literal.toString();
+    if (literal.dataType() instanceof StringType) {
+      return "'"
+              + StringUtils.replace(
```

Comment:
Can we stick to 2 spaces? See "Code style guide" in https://spark.apache.org/contributing.html